### PR TITLE
refactor: use `static` fields

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Replaces the older direct prop assignments on structures with static fields.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Comment by ImRodry** [comment]
Why the name change? Usually constants are written in upper case in JS (https://www.robinwieruch.de/javascript-naming-conventions/)

File: packages/discord.js/src/util/PermissionsBitField.js [hunk 1]
```diff
@@ -64,32 +92,6 @@ class PermissionsBitField extends BitField {
...
-PermissionsBitField.Default = BigInt(104324673);
-
-/**
- * Bitfield representing the permissions required for moderators of stage channels
- * @type {bigint}
- */
-PermissionsBitField.StageModerator =
-  PermissionFlagsBits.ManageChannels | PermissionFlagsBits.MuteMembers | PermissionFlagsBits.MoveMembers;
-
 PermissionsBitField.defaultBit = BigInt(0);
```
Comment (almeidx) [inline packages/discord.js/src/util/PermissionsBitField.js:95]: Remove this

**Review by almeidx — COMMENTED** [review]
https://github.com/discordjs/discord.js/blob/72577c4bfd02524a27afb6ff4aebba9301a690d3/packages/discord.js/src/structures/GuildAuditLogs.js#L525-L526

https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/util/Formatters.js

https://github.com/discordjs/discord.js/blob/72577c4bfd02524a27afb6ff4aebba9301a690d3/packages/discord.js/src/util/Options.js#L174

File: packages/discord.js/src/util/Formatters.js [hunk 2]
```diff
@@ -24,185 +24,185 @@ const {
...
- * @memberof Formatters
- * @param {string} userId The user id to format.
- * @returns {string}
- */
-Formatters.userMention = userMention;
+class Formatters extends null {
+  /**
+   * Formats the content into a block quote. This needs to be at the start of the line for Discord to format it.
+   * @method blockQuote
+   * @memberof Formatters
```
Comment (almeidx) [inline packages/discord.js/src/util/Formatters.js:31]: Are the `@memberof`s still needed?

File: packages/discord.js/src/util/Options.js [hunk 3]
```diff
@@ -153,31 +168,16 @@ class Options extends null {
    * * `GuildChannelManager` - Sweep archived threads
    * * `ThreadManager` - Sweep archived threads
    * <info>If you want to keep default behavior and add on top of it you can use this object and add on to it, e.g.
-   * `makeCache: Options.cacheWithLimits({ ...Options.defaultMakeCacheSettings, ReactionManager: 0 })`</info>
+   * `makeCache: Options.cacheWithLimits({ ...Options.DefaultMakeCacheSettings, ReactionManager: 0 })`</info>
    * @type {Object<string, LimitedCollectionOptions|number>}
    */
-  static get defaultMakeCacheSettings() {
+  static get DefaultMakeCacheSettings() {
```
Comment (almeidx) [inline packages/discord.js/src/util/Options.js:174]: If the casing on these is changed then the `BitField`s `defaultBit` should be changed too

File: packages/discord.js/src/util/Options.js [hunk 4]
```diff
@@ -66,6 +66,21 @@ const Transformers = require('./Transformers');
...
 class Options extends null {
+  /**
+   * The default settings passed to {@link Options.sweepers} (for v14).
+   * The sweepers that this changes are:
+   * * `threads` - Sweep archived threads every hour, removing those archived more than 4 hours ago
+   * <info>If you want to keep default behavior and add on top of it you can use this object and add on to it, e.g.
+   * `sweepers: { ...Options.DefaultSweeperSettings, messages: { interval: 300, lifetime: 600 } })`</info>
+   * @type {SweeperOptions}
+   */
+  static DefaultSweeperSettings = {
```
Comment (almeidx) [inline packages/discord.js/src/util/Options.js:77]: This should probably be a getter like the makeCache one

File: packages/discord.js/typings/index.d.ts [hunk 5]
```diff
@@ -774,8 +774,8 @@ export class ClientUser extends User {
 
 export class Options extends null {
   private constructor();
-  public static defaultMakeCacheSettings: CacheWithLimitsOptions;
-  public static defaultSweeperSettings: SweeperOptions;
+  public static DefaultMakeCacheSettings: CacheWithLimitsOptions;
+  public static DefaultSweeperSettings: SweeperOptions;
```
Comment (almeidx) [inline packages/discord.js/typings/index.d.ts:778]: Should be typed as getters (at least the makeCache one)

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**